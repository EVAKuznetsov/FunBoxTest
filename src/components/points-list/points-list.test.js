import React from 'react'
import PointsList from './points-list'

describe('Tests for Point list',()=>{ 
  const props = {
    points:[
      {
        id:990,
        label:'Первая точка',
        position:[55.75,37.68]        
      },
      {
        id:991,
        label:'Вторая точка',
        position:[55.77,37.61]        
      }
    ],
    delPoint:jest.fn(),
    onChangeOderPoints:jest.fn()
  }
  const pointList = mount(<PointsList {...props} />);
  it('renders pointList',()=>{
    expect(shallowToJson(pointList)).toMatchSnapshot()
  })
  it('Count of points',()=>{
    const items = pointList.find('li.list-group-item');
    expect(items).toHaveLength(props.points.length);
  })
  it('Dellet point',()=>{
    const delBtn = pointList.find('button.del-icon');
    delBtn.first().simulate('click');
    expect(props.delPoint).toBeCalled();
  })
  it('label from props',()=>{
    const items = pointList.find('li.list-group-item');
    expect(items.first().text()).toEqual(props.points[0].label+'x');
  })
})