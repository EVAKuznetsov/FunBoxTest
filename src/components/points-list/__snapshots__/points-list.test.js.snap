// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Point list renders form 1`] = `
<PointsList
  delPoint={[MockFunction]}
  onChangeOderPoints={[MockFunction]}
  points={
    Array [
      Object {
        "id": 990,
        "label": "Первая точка",
        "position": Array [
          55.75,
          37.68,
        ],
      },
      Object {
        "id": 991,
        "label": "Вторая точка",
        "position": Array [
          55.77,
          37.61,
        ],
      },
    ]
  }
>
  <DragDropContext
    onDragEnd={[Function]}
  >
    <ErrorBoundary>
      <App
        onDragEnd={[Function]}
        setOnError={[Function]}
        uniqueId={0}
      >
        <Provider
          context={
            Object {
              "$$typeof": Symbol(react.context),
              "Consumer": Object {
                "$$typeof": Symbol(react.context),
                "_calculateChangedBits": null,
                "_context": [Circular],
              },
              "Provider": Object {
                "$$typeof": Symbol(react.provider),
                "_context": [Circular],
              },
              "_calculateChangedBits": null,
              "_currentRenderer": Object {},
              "_currentRenderer2": null,
              "_currentValue": null,
              "_currentValue2": null,
              "_threadCount": 0,
            }
          }
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
        >
          <ConnectFunction
            direction="vertical"
            droppableId="droppable"
            ignoreContainerClipping={false}
            isCombineEnabled={false}
            isDropDisabled={false}
            type="DEFAULT"
          >
            <Droppable
              direction="vertical"
              droppableId="droppable"
              ignoreContainerClipping={false}
              isCombineEnabled={false}
              isDropDisabled={false}
              placeholder={null}
              shouldAnimatePlaceholder={true}
              snapshot={
                Object {
                  "draggingFromThisWith": null,
                  "draggingOverWith": null,
                  "isDraggingOver": false,
                }
              }
              type="DEFAULT"
              updateViewportMaxScroll={[Function]}
            >
              <ul
                className="list-group"
              >
                <ConnectFunction
                  disableInteractiveElementBlocking={false}
                  draggableId={990}
                  index={0}
                  isDragDisabled={false}
                  key="990"
                  shouldRespectForceTouch={true}
                >
                  <Draggable
                    disableInteractiveElementBlocking={false}
                    draggableId={990}
                    drop={[Function]}
                    dropAnimationFinished={[Function]}
                    index={0}
                    isDragDisabled={false}
                    lift={[Function]}
                    mapped={
                      Object {
                        "combineTargetFor": null,
                        "offset": Object {
                          "x": 0,
                          "y": 0,
                        },
                        "shouldAnimateDisplacement": true,
                        "snapshot": Object {
                          "combineTargetFor": null,
                          "combineWith": null,
                          "draggingOver": null,
                          "dropAnimation": null,
                          "isDragging": false,
                          "isDropAnimating": false,
                          "mode": null,
                        },
                        "type": "SECONDARY",
                      }
                    }
                    move={[Function]}
                    moveByWindowScroll={[Function]}
                    moveDown={[Function]}
                    moveLeft={[Function]}
                    moveRight={[Function]}
                    moveUp={[Function]}
                    shouldRespectForceTouch={true}
                  >
                    <li
                      aria-roledescription="Draggable item. Press space bar to lift"
                      className="list-group-item"
                      data-react-beautiful-dnd-drag-handle="0"
                      data-react-beautiful-dnd-draggable="0"
                      draggable={false}
                      onBlur={[Function]}
                      onDragStart={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                      onTransitionEnd={null}
                      style={
                        Object {
                          "transform": null,
                          "transition": null,
                        }
                      }
                      tabIndex={0}
                    >
                      Первая точка
                      <button
                        className="del-icon"
                        onClick={[Function]}
                      >
                        x
                      </button>
                    </li>
                  </Draggable>
                </ConnectFunction>
                <ConnectFunction
                  disableInteractiveElementBlocking={false}
                  draggableId={991}
                  index={1}
                  isDragDisabled={false}
                  key="991"
                  shouldRespectForceTouch={true}
                >
                  <Draggable
                    disableInteractiveElementBlocking={false}
                    draggableId={991}
                    drop={[Function]}
                    dropAnimationFinished={[Function]}
                    index={1}
                    isDragDisabled={false}
                    lift={[Function]}
                    mapped={
                      Object {
                        "combineTargetFor": null,
                        "offset": Object {
                          "x": 0,
                          "y": 0,
                        },
                        "shouldAnimateDisplacement": true,
                        "snapshot": Object {
                          "combineTargetFor": null,
                          "combineWith": null,
                          "draggingOver": null,
                          "dropAnimation": null,
                          "isDragging": false,
                          "isDropAnimating": false,
                          "mode": null,
                        },
                        "type": "SECONDARY",
                      }
                    }
                    move={[Function]}
                    moveByWindowScroll={[Function]}
                    moveDown={[Function]}
                    moveLeft={[Function]}
                    moveRight={[Function]}
                    moveUp={[Function]}
                    shouldRespectForceTouch={true}
                  >
                    <li
                      aria-roledescription="Draggable item. Press space bar to lift"
                      className="list-group-item"
                      data-react-beautiful-dnd-drag-handle="0"
                      data-react-beautiful-dnd-draggable="0"
                      draggable={false}
                      onBlur={[Function]}
                      onDragStart={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                      onTransitionEnd={null}
                      style={
                        Object {
                          "transform": null,
                          "transition": null,
                        }
                      }
                      tabIndex={0}
                    >
                      Вторая точка
                      <button
                        className="del-icon"
                        onClick={[Function]}
                      >
                        x
                      </button>
                    </li>
                  </Draggable>
                </ConnectFunction>
                <AnimateInOut
                  on={null}
                  shouldAnimate={true}
                />
              </ul>
            </Droppable>
          </ConnectFunction>
        </Provider>
      </App>
    </ErrorBoundary>
  </DragDropContext>
</PointsList>
`;

exports[`Tests for Point list renders pointList 1`] = `
<PointsList
  delPoint={[MockFunction]}
  onChangeOderPoints={[MockFunction]}
  points={
    Array [
      Object {
        "id": 990,
        "label": "Первая точка",
        "position": Array [
          55.75,
          37.68,
        ],
      },
      Object {
        "id": 991,
        "label": "Вторая точка",
        "position": Array [
          55.77,
          37.61,
        ],
      },
    ]
  }
>
  <DragDropContext
    onDragEnd={[Function]}
  >
    <ErrorBoundary>
      <App
        onDragEnd={[Function]}
        setOnError={[Function]}
        uniqueId={0}
      >
        <Provider
          context={
            Object {
              "$$typeof": Symbol(react.context),
              "Consumer": Object {
                "$$typeof": Symbol(react.context),
                "_calculateChangedBits": null,
                "_context": [Circular],
              },
              "Provider": Object {
                "$$typeof": Symbol(react.provider),
                "_context": [Circular],
              },
              "_calculateChangedBits": null,
              "_currentRenderer": Object {},
              "_currentRenderer2": null,
              "_currentValue": null,
              "_currentValue2": null,
              "_threadCount": 0,
            }
          }
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
        >
          <ConnectFunction
            direction="vertical"
            droppableId="droppable"
            ignoreContainerClipping={false}
            isCombineEnabled={false}
            isDropDisabled={false}
            type="DEFAULT"
          >
            <Droppable
              direction="vertical"
              droppableId="droppable"
              ignoreContainerClipping={false}
              isCombineEnabled={false}
              isDropDisabled={false}
              placeholder={null}
              shouldAnimatePlaceholder={true}
              snapshot={
                Object {
                  "draggingFromThisWith": null,
                  "draggingOverWith": null,
                  "isDraggingOver": false,
                }
              }
              type="DEFAULT"
              updateViewportMaxScroll={[Function]}
            >
              <ul
                className="list-group"
              >
                <ConnectFunction
                  disableInteractiveElementBlocking={false}
                  draggableId={990}
                  index={0}
                  isDragDisabled={false}
                  key="990"
                  shouldRespectForceTouch={true}
                >
                  <Draggable
                    disableInteractiveElementBlocking={false}
                    draggableId={990}
                    drop={[Function]}
                    dropAnimationFinished={[Function]}
                    index={0}
                    isDragDisabled={false}
                    lift={[Function]}
                    mapped={
                      Object {
                        "combineTargetFor": null,
                        "offset": Object {
                          "x": 0,
                          "y": 0,
                        },
                        "shouldAnimateDisplacement": true,
                        "snapshot": Object {
                          "combineTargetFor": null,
                          "combineWith": null,
                          "draggingOver": null,
                          "dropAnimation": null,
                          "isDragging": false,
                          "isDropAnimating": false,
                          "mode": null,
                        },
                        "type": "SECONDARY",
                      }
                    }
                    move={[Function]}
                    moveByWindowScroll={[Function]}
                    moveDown={[Function]}
                    moveLeft={[Function]}
                    moveRight={[Function]}
                    moveUp={[Function]}
                    shouldRespectForceTouch={true}
                  >
                    <li
                      aria-roledescription="Draggable item. Press space bar to lift"
                      className="list-group-item"
                      data-react-beautiful-dnd-drag-handle="0"
                      data-react-beautiful-dnd-draggable="0"
                      draggable={false}
                      onBlur={[Function]}
                      onDragStart={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                      onTransitionEnd={null}
                      style={
                        Object {
                          "transform": null,
                          "transition": null,
                        }
                      }
                      tabIndex={0}
                    >
                      Первая точка
                      <button
                        className="del-icon"
                        onClick={[Function]}
                      >
                        x
                      </button>
                    </li>
                  </Draggable>
                </ConnectFunction>
                <ConnectFunction
                  disableInteractiveElementBlocking={false}
                  draggableId={991}
                  index={1}
                  isDragDisabled={false}
                  key="991"
                  shouldRespectForceTouch={true}
                >
                  <Draggable
                    disableInteractiveElementBlocking={false}
                    draggableId={991}
                    drop={[Function]}
                    dropAnimationFinished={[Function]}
                    index={1}
                    isDragDisabled={false}
                    lift={[Function]}
                    mapped={
                      Object {
                        "combineTargetFor": null,
                        "offset": Object {
                          "x": 0,
                          "y": 0,
                        },
                        "shouldAnimateDisplacement": true,
                        "snapshot": Object {
                          "combineTargetFor": null,
                          "combineWith": null,
                          "draggingOver": null,
                          "dropAnimation": null,
                          "isDragging": false,
                          "isDropAnimating": false,
                          "mode": null,
                        },
                        "type": "SECONDARY",
                      }
                    }
                    move={[Function]}
                    moveByWindowScroll={[Function]}
                    moveDown={[Function]}
                    moveLeft={[Function]}
                    moveRight={[Function]}
                    moveUp={[Function]}
                    shouldRespectForceTouch={true}
                  >
                    <li
                      aria-roledescription="Draggable item. Press space bar to lift"
                      className="list-group-item"
                      data-react-beautiful-dnd-drag-handle="0"
                      data-react-beautiful-dnd-draggable="0"
                      draggable={false}
                      onBlur={[Function]}
                      onDragStart={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onMouseDown={[Function]}
                      onTouchStart={[Function]}
                      onTransitionEnd={null}
                      style={
                        Object {
                          "transform": null,
                          "transition": null,
                        }
                      }
                      tabIndex={0}
                    >
                      Вторая точка
                      <button
                        className="del-icon"
                        onClick={[Function]}
                      >
                        x
                      </button>
                    </li>
                  </Draggable>
                </ConnectFunction>
                <AnimateInOut
                  on={null}
                  shouldAnimate={true}
                />
              </ul>
            </Droppable>
          </ConnectFunction>
        </Provider>
      </App>
    </ErrorBoundary>
  </DragDropContext>
</PointsList>
`;
